core/src/main/java/org/apache/iceberg/rest/requests/RegisterTableRequest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.RESTRequest;
+
+public class RegisterTableRequest implements RESTRequest {
```

Comment:
@rdblue I'm not sure if we decided on which pattern we'd want to use with new request/response classes:
1) class with a public constructor and no explicit JSON parser + manually written builder
2) Immutable class (similar to `ReportMetricsRequest`) with a separate JSON parser

Reply:
@krvikash since we don't want to rely on reflection (anymore) and want to be explicit, could you please add a separate JSON parser for that request class? This would look similar to `CommitTransactionRequestParser`